plugin-build/src/main/kotlin/io/sentry/android/gradle/AppConfig.kt

```diff
@@ -69,6 +78,76 @@ fun AppExtension.configure(
             sentryOrg,
             sentryProject
         )
+
+        variant.configureDebugMetaPropertiesTask(
+            project,
+            this,
+            mergeAssetsDependants,
+            tasksGeneratingProperties
+        )
+    }
+}
+
+private fun ApplicationVariant.configureDebugMetaPropertiesTask(project: Project, appExtension: AppExtension, dependants: Set<TaskProvider<out Task>?>, tasksGeneratingProperties: List<TaskProvider<*>>) {
+    if (isAGP74) {
+        project.logger.info {
+            "Not configuring deprecated AppExtension for ${AgpVersions.CURRENT}, " +
+                "new AppComponentsExtension will be configured"
+        }
+        project.logger.error("not hello 70 ${AgpVersions.CURRENT}")
+    } else {
+        project.logger.error("hello 70 ${AgpVersions.CURRENT}")
+        val paths = OutputPaths(project, name)
+        val variant = AndroidVariant70(this)
+        val taskSuffix = name.capitalized
+        val outputDir = project.layout.buildDirectory.dir(
+            "generated${sep}assets${sep}sentry${sep}debug-meta-properties${sep}$name")
+        val generateDebugMetaPropertiesTask = SentryGenerateDebugMetaPropertiesTask.register(
+            project,
+            tasksGeneratingProperties,
+            outputDir,
+            taskSuffix
+        )
+
+        generateDebugMetaPropertiesTask.setupMergeAssetsDependencies(dependants)
+        generateDebugMetaPropertiesTask.hookWithPackageTasks(project, variant)
+        appExtension.sourceSets.getByName(name).assets.srcDir(
+            generateDebugMetaPropertiesTask.flatMap { it.output }
+        )
+    }
+}
+
+private fun ApplicationVariant.configureSourceBundleTasks(project: Project, extension: SentryPluginExtension, appExtension: AppExtension, cliExecutable: String, dependants: Set<TaskProvider<out Task>?>): SourceContext.SourceContextTasks? {
+    if (isAGP74) {
+        project.logger.info {
+            "Not configuring deprecated AppExtension for ${AgpVersions.CURRENT}, " +
+                "new AppComponentsExtension will be configured"
+        }
+        project.logger.error("not hello 70 ${AgpVersions.CURRENT}")
+        return null
+    } else if (extension.includeSourceBundle.get()) {
+        project.logger.error("hello 70 ${AgpVersions.CURRENT}")
+        val paths = OutputPaths(project, name)
+        val variant = AndroidVariant70(this)
+        val taskSuffix = name.capitalized
+        val sourceFiles = this.sourceSets.flatMap { it.javaDirectories.flatMap { project.fileTree(it) } }
```

Comment:
does this break caching; should we use provider if so how?

Reply:
added that we don't include kotlin sources to https://github.com/getsentry/sentry-java/issues/633#issuecomment-1465599120 as known limitation for now. Will take another look later on.